File: authorization/src/main/kotlin/com/example/authorization/controller/AuthorizationController.java
package com.example.authorization.controller;

import com.example.authorization.service.*;
import lombok.*;
import org.springframework.http.*;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
@RequiredArgsConstructor
public class AuthorizationController {

    private final AuthorizationService authorizationService;

    @PostMapping("/register")
    @ResponseStatus(HttpStatus.CREATED)
    public void registerUser(@RequestBody UserRequest userRequest) {
        authorizationService.registerUser(userRequest);
    }

    @PostMapping("/login")
    @ResponseStatus(HttpStatus.ACCEPTED)
    public String loginUser(@RequestBody UserRequest userRequest) {
        return authorizationService.loginUser(userRequest);
    }

    @GetMapping("/user-info")
    @ResponseStatus(HttpStatus.OK)
    public UserResponse getUserDetailsAfterLogin(@RequestHeader("Authorization") String jwt) {
        return authorizationService.getUserDetails(jwt);
    }
}

File: authorization/src/main/kotlin/com/example/authorization/controller/AuthorizationController.kt
package com.example.authorization.controller

import com.example.authorization.service.AuthorizationService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class AuthorizationController(private val authorizationService: AuthorizationService) {

    @PostMapping("/register")
    @ResponseStatus(HttpStatus.CREATED)
    fun registerUser(@RequestBody registerRequest: RegisterRequest) =
        authorizationService.registerUser(registerRequest)

    @PostMapping("/login")
    @ResponseStatus(HttpStatus.ACCEPTED)
    fun loginUser(@RequestBody loginRequest: LoginRequest) =
        authorizationService.loginUser(loginRequest)

    @GetMapping("/user-info")
    @ResponseStatus(HttpStatus.OK)
    fun getUserDetailsAfterLogin(@RequestHeader("Authorization") jwt: String) =
        authorizationService.getUserDetails(jwt)
}
